crypto/cms/cms_lcl.h

```diff
@@ -317,7 +317,7 @@ struct CMS_OtherKeyAttribute_st {
 
 /* ESS structures */
 
-# ifdef HEADER_X509V3_H
+# ifdef OPENSSL_X509V3_H
```

Comment:
a rather funny thing to do in a header file, one should simply include the necessary files...

Reply:
That's a hard riddle: It took me some time to understand what the original intention for adding this guard, but now I think I found some clues:

There are actually two `HEADER_X509V3_H` guards, one in `cms_lcl.h` and one in `cms.h`.

The first one in `cms_lcl.h` was added in commit be86c7fc8784f785b20bad50f71327a57068a565:

https://github.com/openssl/openssl/blob/be86c7fc8784f785b20bad50f71327a57068a565/crypto/cms/cms_lcl.h#L380-L399

Looking at line 56 of `cms_asn1.c`

https://github.com/openssl/openssl/blob/be86c7fc8784f785b20bad50f71327a57068a565/crypto/cms/cms_asn1.c#L54-L58

which was [added in the same commit](https://github.com/openssl/openssl/commit/be86c7fc8784f785b20bad50f71327a57068a565#diff-9a1346101607e9184312ccb7b51fcdb2R56), it occurred to me that the original purpose was probably to hide the declarations for everybody except for `cms_asn1.c`.

The second one in `cms.h` was added two days later in f4cc56f494b3f42642ea3b322e57737da80f889a, probably for a similar reason:

https://github.com/openssl/openssl/blob/f4cc56f494b3f42642ea3b322e57737da80f889a/crypto/cms/cms.h#L288-L302

If this ever made sense, the reason for doing it has probably vanished long time ago  (maybe a no-x509v3 option?). So unless anybody of you gives me a better explanation, I'll just assume that these guards can safely be removed.